static/js/components/EducationForm.js

```diff
@@ -334,16 +341,21 @@ class EducationForm extends ProfileFormFields {
         </Card>;
       });
     } else if (profile !== undefined) {
```

Comment:
My point more broadly is the the check that you have up in `renderEducationLevelEntries` should be done in this function. `renderEducationLevelEntries` should do what it says and not have control flow in it, and the logic for when to render things and when not to should be in one place - in this function (`renderCard`).

Basically we want to do this, if I'm understanding you correctly:

```js
} else if (profile !== undefined)
  if (!isProfileOfLoggedinUser(profile) && (!profile.education || profile.education.length === 0)) {
    return null;
  }
  return <Card ... />;
```

If I'm understanding you correctly there are 4 cases:

1. profile not logged in, education present
2. profile not logged in, education not present
3. profile logged in, education present
4. profile logged in, education not present
  
With this setup in case 1, 3, and 4 we would render education entries, and we would omit them in case 2.

Isn't that what we're shooting for? If not let me know what I'm missing.
 So in this function we check for all the conditions where we would want to hide the education entries (if the profile is not logged in and there aren't education entries on it). In all other cases (education entries present on non-logged in user